### DISCUSSION TITLE
How to pass conditions to solve()

### DISCUSSION THREAD

**Discussion by yanivabir** [mwshinn/PyDDM#49] [Ask for help] [answered]
Hi,

I'm trying to simulate data for the simple Roitman DDM model (the second model in the getting started tutorial, with static bounds).

To use model.solve(), I need to pass conditions. I can pass one condition value just fine: `model.solve(conditions={"coh": .3})`, but I can't figure out the correct format for passing more than one condition. And I can't seem to find it in the docs. 

Any clues?

Thanks!
Yaniv

**Comment by mwshinn** [comment, marked as answer]
It is a Python dictionary, so in this example, if you had a condition "contrast", you could do:

`model.solve(conditions={"coh": .3, "contrast": .5})`

If you want multiple coherences in one call to solve, this is an underspecified problem because you need to specify the number of trials of each coherence.  You can do this with [solve_partial_conditions](https://pyddm.readthedocs.io/en/latest/apidoc/utils.html#ddm.functions.solve_partial_conditions):

To use a sample:

`ddm.solve_partial_conditions(model, sample=sample)`

For equal ratios of all conditions you specify:

`ddm.solve_partial_conditions(model, conditions={"coh": [.3, .6, .9]})`

**Reply by yanivabir** [reply]
cool, thanks!

running solve_partial_conditions on a sample, I then try to resample to get simulated data. But the returned sample has no condition field when I use to_pandas_dataframe on it. 
How would I get a fake dataset equivalent to the one I used solve_partial_conditions on?

**Reply by mwshinn** [reply]
To do this you would have to do it manually, since there is no convenience function to accomplish this.  By that, I mean: find every set of conditions which is used (e.g. through the condition_combinations method of Sample) and then find how many trials there are which use that combination of conditions (e.g. by finding the length after using the subset method of Sample).  Then, simulate that many of each set of conditions.  You can combine sample objects using the + sign to make this easier.

**Reply by yanivabir** [reply]
Thanks! and the + sign does make this much easier!
